File: 2017/Lecture 3/GitHubProjects_2017_03/app/src/main/java/com/avast/example/android/github/net/GitHubApi.java
package com.avast.example.android.github.net;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Headers;
import retrofit2.http.Path;

import com.avast.example.android.github.model.Repo;
import com.avast.example.android.github.model.User;

/**
 * Created by lukas on 11/8/17.
 */

public interface GitHubApi {

    @Headers({
            "Accept: application/vnd.github.v3.full+json",
            "User-Agent: Retrofit-Sample-App"
    })
    @GET("/users/{username}")
    Call<User> getUser(@Path("username") String username);

    @Headers({
            "Accept: application/vnd.github.v3.full+json",
            "User-Agent: Retrofit-Sample-App"
    })
    @GET("/users/{username}/repos")
    Call<List<Repo>> getUserRepo(@Path("username") String username);

    @Headers({
            "Accept: application/vnd.github.v3.full+json",
            "User-Agent: Retrofit-Sample-App"
    })
    @GET("/repos/{owner}/{repository}/contributors")
    Call<List<User>> getRepoContributors(@Path("owner") String owner,
            @Path("repository") String repoName);


}

File: 2017/Lecture 6/App/app/src/main/java/com/avast/android/githubbrowser/repository/GithubApi.kt
package com.avast.android.githubbrowser.repository

import com.avast.android.githubbrowser.pojo.Repository
import com.avast.android.githubbrowser.pojo.User
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

/**
 * Created by lukas on 12/21/17.
 */
interface GithubApi {

    companion object {
        val GITHUB_API_URL = "https://api.github.com"
    }

    @GET("users")
    fun getUsers(@Query("since") id: String): Call<List<User>>

    @GET("users/{user}")
    fun user(@Path("user") user: String): Call<User>

    @GET("users/{user}/repos")
    fun listRepos(@Path("user") user: String): Call<List<Repository>>

    @GET("repos/{owner}/{repoName}/contributors")
    fun repoContributors(@Path("owner") owner: String, @Path("repoName") repoName: String): Call<List<User>>

}